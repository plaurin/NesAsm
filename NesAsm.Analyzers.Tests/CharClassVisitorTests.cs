using ApprovalTests;
using ApprovalTests.Reporters;
using NesAsm.Analyzers.Visitors;

namespace NesAsm.Analyzers.Tests;

[UseReporter(typeof(VisualStudioReporter))]
public class CharClassVisitorTests
{
    [Fact]
    public void Test()
    {
        var result = CharClassVisitor.Process(@"C:\Users\pasca\Dev\GitHub\NesAsm\NesAsm.Example\game1.png");

        Approvals.Verify(result);
    }

    [Fact]
    public void TestImportCharScript()
    {
        TestScript("ImportCharDefinition.cs");
    }

    private static void TestScript(params string[] filenames)
    {
        var generator = new CharGenerator();

        var sourcePaths = filenames.Select(filename => $@"..\..\..\TestFiles\{filename}").ToArray();

        GeneratorTestUtilities.TestGenerator(generator, sourcePaths);
    }
}