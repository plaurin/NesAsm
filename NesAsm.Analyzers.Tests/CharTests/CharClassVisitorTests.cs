using ApprovalTests.Reporters;

namespace NesAsm.Analyzers.Tests.CharTests;

[UseReporter(typeof(VisualStudioReporter))]
public class CharClassVisitorTests
{
    [Fact]
    public void TestImportCharScript()
    {
        TestScript("ImportCharDefinition.cs");
    }

    [Fact]
    public void TestInvalidSizeCharScript()
    {
        TestScript("InvalidSize.cs");
    }

    [Fact]
    public void TestMoreThanFourPalettesCharScript()
    {
        TestScript("MoreThanFourPalettes.cs");
    }

    [Fact]
    public void TestMoreThanThreeNonDefaultColorCharScript()
    {
        TestScript("MoreThanThreeNonDefaultColor.cs");
    }

    [Fact]
    public void TestColorMismatchCharScript()
    {
        TestScript("ColorMismatch.cs");
    }

    private static void TestScript(params string[] filenames)
    {
        var generator = new CharGenerator();

        var sourcePaths = filenames.Select(filename => $@"..\..\..\CharTests\TestFiles\{filename}").ToArray();

        GeneratorTestUtilities.TestGenerator(generator, sourcePaths);
    }
}