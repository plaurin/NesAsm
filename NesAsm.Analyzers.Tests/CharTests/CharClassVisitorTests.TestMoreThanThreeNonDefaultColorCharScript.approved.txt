- Main diagnostics:
..\..\..\CharTests\TestFiles\MoreThanThreeNonDefaultColor.cs(6,2): warning NA1003: Tile index 1 contains 7 non default colors (should be 3 maximum in a color palette), excess colors will be ignored
- SyntaxTrees Count: 2
- Output diagnostics:
- RunResult.GeneratedTrees Count: 1
- RunResults diagnostics:
..\..\..\CharTests\TestFiles\MoreThanThreeNonDefaultColor.cs(6,2): warning NA1003: Tile index 1 contains 7 non default colors (should be 3 maximum in a color palette), excess colors will be ignored
- RunResult[0].Generator: Microsoft.CodeAnalysis.IncrementalGeneratorWrapper
- RunResult[0] diagnostics:
..\..\..\CharTests\TestFiles\MoreThanThreeNonDefaultColor.cs(6,2): warning NA1003: Tile index 1 contains 7 non default colors (should be 3 maximum in a color palette), excess colors will be ignored
- RunResult[0].GeneratedSources Count: 1
- RunResult[0].Exception: 
- RunResult[0].GeneratedSources[0].HintName: Asm.MoreThanThreeNonDefaultColor.cs
- RunResult[0].GeneratedSources[0].SourceText:
/* Generator Asm code in file Asm.MoreThanThreeNonDefaultColor.s
; Auto generated code using the NesAsm project
.scope MoreThanThreeNonDefaultColor

.segment "CHARS"

  ; Sprite Tile 0
  .byte 0, 0, 0, 0, 0, 0, 0, 0
  .byte 0, 0, 0, 0, 0, 0, 0, 0

  ; Sprite Tile 1
  .byte %00001000
  .byte %00001100
  .byte %00001100
  .byte %11000000
  .byte %11000000
  .byte %00000000
  .byte %00000000
  .byte %00000000

  .byte %00000000
  .byte %00000000
  .byte %00000010
  .byte %11001111
  .byte %11001111
  .byte %00000010
  .byte %00000000
  .byte %00000000


.segment "CODE"

sprite_palettes:
  ; Sprite Palette 0
  .byte 0F, 05, 22, 29

.endscope


*/
